catalyst/metrics/ndcg.py

```diff
@@ -0,0 +1,79 @@
+"""
+Discounted Cumulative Gain metrics:
+    * :func:`dcg`
+    * :func:`ndcg`
+"""
+from typing import Sequence
+
+import torch
+
+
+def dcg(
+    outputs: torch.Tensor, targets: torch.Tensor, 
+    gain_function=lambda x: torch.pow(2, x) - 1, k=100
+) -> torch.Tensor:
+    """
+    Computes DCG@topk for the specified values of `topk`.
+    Reference:
+        https://en.wikipedia.org/wiki/Discounted_cumulative_gain
+
+    Args:
+        outputs (torch.Tensor): model outputs, logits
+            with shape [batch_size; slate_length]
+        targets (torch.Tensor): ground truth, labels
+            with shape [batch_size; slate_length]
+        gain_function: 
+            callable, gain function for the ground truth labels. 
+            on the deafult, the torch.pow(2, x) - 1 function used
+            to get emphasise on the retirvng the revelant documnets.
+        k (int):
+            Parameter fro evaluation on top-k items
+
+    Returns:
+        list with computed dcg@topk
+    """
+    k = min(outputs.size(1), k)
```

Comment:
should we assert `outputs.size(1) > k`?

Reply:
or warning?

Reply:
Will be fixed in PR for preprocessing pipeline